lib/versioning/composer/index.spec.ts

```diff
@@ -312,6 +312,22 @@ describe('semver.getNewValue()', () => {
         newVersion: '3.1.0',
       })
     ).toEqual('~1.2 || ~2.0 || ~3.0');
+    expect(
+      semver.getNewValue({
+        currentValue: '^1.2',
+        rangeStrategy: 'widen',
+        currentVersion: '1.2.0',
+        newVersion: '2.0.0',
+      })
+    ).toEqual('^1.2 || ^2.0');
+    expect(
+      semver.getNewValue({
+        currentValue: '~1.2',
+        rangeStrategy: 'widen',
+        currentVersion: '1.2.0',
+        newVersion: '2.4.0',
+      })
+    ).toEqual('~1.2 || ~2.0');
```

Comment:
What about this tests?
```ts
    expect(
      semver.getNewValue({
        currentValue: '~1.2',
        rangeStrategy: 'widen',
        currentVersion: '1.2.0',
        newVersion: '1.9.0',
      })
    ).toEqual('?');
    expect(
      semver.getNewValue({
        currentValue: '^1.2',
        rangeStrategy: 'widen',
        currentVersion: '1.2.0',
        newVersion: '1.9.0',
      })
    ).toEqual('?');
```

Reply:
good point, I've added a case to not modify if the new version satisfies the existing for widening.
I think both those ~ and ^ cases should be the same?